src/digitalmarketplace/components/list-input/list-input.js

```diff
@@ -0,0 +1,211 @@
+var getSibling = function (direction, elem, selector) {
+  // Get the next sibling element
+  var sibling = (direction === 'next') ? elem.nextElementSibling : elem.previousElementSibling
+
+  // If there's no selector, return the first sibling
+  if (!selector) return sibling
+
+  // If the sibling matches our selector, use it
+  // If not, jump to the next sibling and continue the loop
+
+  // For IE10,11 (Polyfill for matches)
+  if (!Element.prototype.matches) {
+    Element.prototype.matches = Element.prototype.msMatchesSelector ||
+                                Element.prototype.webkitMatchesSelector
+  }
+  while (sibling) {
+    if (sibling.matches(selector)) return sibling
+    sibling = (direction === 'next') ? sibling.nextElementSibling : sibling = sibling.previousElementSibling
+  }
+}
+
+function ListInput ($module) {
+  this.$module = $module
+  this.$allVisibleItems = $module.querySelectorAll('.dm-list-input__item:not(.dm-list-input__item--hidden)')
+  this.$addAnotherButton = $module.querySelector('.dm-list-input__item-add')
+  this.$allItems = $module.querySelectorAll('.dm-list-input__item')
+  this.visibleItems = 0
+
+  this.itemContainerClass = 'dm-list-input__item-container'
+  this.hiddenItemClass = 'dm-list-input__item--hidden'
+  this.visibleItemClass = 'dm-list-input__item--visible'
+  this.itemInputClass = 'dm-list-input__item-input'
+  this.itemErrorClass = 'dm-list-input__item--error'
+  this.itemErrorMsgClass = 'govuk-error-message'
+  this.itemCounterClass = 'dm-list-input__counter'
+  this.inputErrorClass = 'govuk-input--error'
+  this.removeButtonClass = 'dm-list-input__item-remove'
+  this.hiddenRemoveButtonClass = 'dm-list-input__item-remove--hidden'
+
+  this.formgroupErrorClass = 'govuk-form-group--error'
```

Comment:
Can we make this `formGroupErrorClass` for consistency with the other variables?